src/wallet/rpcwallet.cpp

```diff
@@ -3408,10 +3405,6 @@ UniValue z_sendmany(const UniValue& params, bool fHelp)
             }
         }
 
-        if (setAddress.count(address))
```

Comment:
I'm not sure it's necessary, but is there value in only _allowing_ duplicates when each instance of the address includes memo data? 

If so, it seems like it would be simple to keep this block but move it down before line 3417 into an isNull memo check. Something like:
```
if (memoValue.isNull() && setAddress.count(address)) {
    throw JSONRPCError(RPC_INVALID_PARAMETER, string("Invalid parameter, missing memo data, duplicated address: ") + address);
}
setAddress.insert(address);
```

Reply:
@joshgentry interesting idea, sounds like it needs an RPC test 😈 

@radix42 your thoughts on requiring memo keys when using duplicate addresses? 

What if I want to send an amount=0 memo and also 5HUSH to the same zaddr in one transaction as different transaction outputs (TXOs) ? Upstream forbids that, should we?

Reply:
I don't have a strong opinion on whether we should allow a non-memo payment + non-payment memo to the same address. It seems like for optimization's sake (which sounds like where this decision originally came from), it would be best to encourage them in the same output.

On the surface level, this optimization seems to be largely superficial. To avoid spam, we might actually want to validate that each output memo was "full" (or full-ish) except for the last one to that address. Otherwise, it would be easy to send a bunch of memo data with one byte of data and bloat the transaction.

Reply:
@joshgentry all memos are "full", if unused, they are full of zero bytes. They take up the same amount of space whether used or not, otherwise, people could detect when people use memos and when they don't